src/workers/sketchfab-zip.worker.js

```diff
@@ -5,13 +5,13 @@ async function fetchZipAndGetBlobs(src) {
   return zip.load().then(() => {
     // Rewrite any url references in the GLTF to blob urls
     const fileMap = {};
-    for (const fileName in zip.files) {
-      fileMap[fileName] = zip.extractAsBlobUrl(fileName);
-    }
-
+    const rewriteUri = uri => {
+      fileMap[uri] = fileMap[uri] || zip.extractAsBlobUrl(uri);
```

Comment:
Note to other reviewers: I haven't checked the `ZipLoader` API yet but I assume the return value from `zip.extractAsBlobUrl(uri)` is not `null` or emptry strings.

Reply:
Checking [ziploader](https://github.com/yomotsu/ZipLoader/blob/master/src/ZipLoader.js#L89) It always returns a blob url. Also it already keeps track of extracted uris so we may simplify here just calling the ziploader method

Reply:
Thanks for checking.

> Also it already keeps track of extracted uris so we may simplify here just calling the ziploader method

The current `fileMap[uri] = fileMap[uri] || zip.extractAsBlobUrl(uri);` is ok to me. `it already keeps track of extracted uris` is ZipLoader internal design. There might be a chance they ZipLoader will change it in the future version.